### PR TITLE
Update to Pants 1.29.0

### PR DESCRIPTION
**Pull request by slvrtrn**
Addresses https://github.com/scalapb/scalapb-pants/issues/14

* Bumped Scalapb* related dependencies 0.9.4 -> 0.10.1
* Pants 1.21 -> 1.29.0
* Pants INI -> TOML
* Fixed `twitter.common.collections import OrderedSet` import issue (see below)
* Fixed examples (incorrect packages, etc...)

------------------------------------------------------------------------------

There are some things that I would like to discuss though

- do we need to mimic that OrderedSet behavior? I've decided to just make a simple set, then convert it to list and have it sorted when everything is done. But is the order important there?
- There are some build warnings I was unable to resolve:

The first one:
```
18:38:56 00:02     [parse]16:38:56:702 [WARN] /Users/serge/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0_py36/lib/python3.6/site-packages/pants/source/source_root.py:109: DeprecationWarning: DEPRECATED: the *_root_patterns and *_roots options will be removed in version 1.30.0.dev0.
  Explicitly list your source roots with the `root_patterns` option in the [source] scope. See https://pants.readme.io/docs/source-roots. See your current roots with `./pants roots`.
  self._trie = None if self._pattern_matcher.get_patterns() else source_root_config.create_trie()  # type: ignore[union-attr]
```

The second one:

```
16:38:56:650 [WARN] /Users/serge/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0_py36/lib/python3.6/site-packages/pants/engine/internals/native.py:489: DeprecationWarning: DEPRECATED: Custom V1 targets not having a Target API binding will be removed in version 1.30.0.dev0.
  You have some custom target types that will soon need light-weight bindings for the new and more powerful Target API to continue working properly with parts of Pants, like the `list` and `filedeps` goals.

See https://groups.google.com/forum/#!topic/pants-devel/WsRFODRLVZI for more information and instructions.

(Could not find bindings for: ['scalapb_library'])
  res = c.from_value(func[0]).send(c.from_value(arg[0]))
```

the examples are working fine though (if called from the console using `./pants run examples/simple:main`)

Also, this project cannot be imported into IDEA 2020.1.3 with the latest Pants build system support plugin, it says:
```
               FAILURE
FAILURE: Modulizable targets must not contain any synthetic target, but in this case the following synthetic targets depend on other non-synthetic modules:
{ScalaLibrary(.pants.d/gen/scalapb-gen/9c44cc2743fa/examples.client.client_proto/current:examples.client.client_proto), ScalaLibrary(.pants.d/gen/scalapb-gen/9c44cc2743fa/examples.grpc.service_proto/current:examples.grpc.service_proto), ScalaLibrary(.pants.d/gen/scalapb-gen/9c44cc2743fa/examples.simple.simple_proto/current:examples.simple.simple_proto), ScalaLibrary(.pants.d/gen/scalapb-gen/9c44cc2743fa/examples.simple.dep_proto/current:examples.simple.dep_proto)}
One approach that may help is to reduce the scope of the import to further avoid synthetic targets.
```

so I had to go "blindly" without any IDE support... Anyway, plugin seems to be working now.

### CODE REVIEW

File: 3rdparty/jvm/BUILD [hunk 1]
```diff
@@ -12,6 +12,6 @@ jar_library(name = 'protobuf-runtime',
 
 jar_library(name = 'scalapb-runtime-grpc',
   jars = [
-    jar('com.thesamet.scalapb', 'scalapb-runtime-grpc_2.12', '0.9.4')
+    jar('com.thesamet.scalapb', 'scalapb-runtime-grpc_2.12', '0.10.1')
```
Comment (thesamet) [inline 3rdparty/jvm/BUILD]: Can we move all the way to latest (0.10.7)?

File: 3rdparty/jvm/BUILD [hunk 2]
```diff
@@ -1,6 +1,6 @@
 jar_library(name = 'scalapb-runtime',
   jars = [
-    jar('com.thesamet.scalapb', 'scalapb-runtime_2.12', '0.9.4')
+    jar('com.thesamet.scalapb', 'scalapb-runtime_2.12', '0.10.1')
```
Comment (thesamet) [inline 3rdparty/jvm/BUILD]: Can we move all the way to latest (0.10.7)?

File: 3rdparty/jvm/BUILD (hunk 1 above)
Comment (slvrtrn) [reply 3rdparty/jvm/BUILD]: Oops! sorry. I've checked the website https://scalapb.github.io/scalapbc.html and saw `0.10.1` there. Should've checked maven repo instead. Now it is fixed. I've also tried to resolve Travis issues and explicitly set the JVM platform to 11 in both Pants TOML and Travis config.

**Comment by slvrtrn** [comment]
@thesamet could you please check it one more time? Scalapbc runtime is the latest now, plus Travis build is fixed as well.

File: BUILD.tools [hunk 3]
```diff
@@ -1,5 +1,5 @@
 jar_library(name = 'scalapbc',
             jars = [
-              scala_jar('com.thesamet.scalapb', 'scalapbc', '0.9.4'),
+              scala_jar('com.thesamet.scalapb', 'scalapbc', '0.10.1'),
```
Comment (thesamet) [inline BUILD.tools]: One more version update

**Comment by thesamet** [comment]
Thank you!

**Pull request merged**